config-backend.js

```diff
@@ -7,5 +7,136 @@ module.exports = {
+            message: 'Hex Arch - Cannot import from application in the infrastructure layer',
+          },
+          {
+            target: './src',
+            from: './test',
+            message: 'Test code - Cannot import from test in the src code',
+          },
+          {
+            target: './test/domain',
+            from: './src/infrastructure',
+            message: 'Hex: domain layer cannot know about the infrastructure layer, even in tests',
+          },
+          {
+            target: './test/domain',
+            from: './src/configuration',
+            message: 'Hex: domain layer cannot know about the configuration layer, even in tests',
+          },
+          {
+            target: './test/domain',
+            from: './src/application',
+            message: 'Hex: domain layer cannot know about the application layer, even in tests',
+          },
+          {
+            target: './test/infrastructure',
+            from: './src/application',
+            message: 'Hex: infrastructure layer cannot know about the application layer, even in tests',
+          },
+          {
+            target: './test/application',
+            from: './src/infrastructure',
+            message: 'Hex: application layer cannot know about the infrastructure layer, even in tests',
+          },
+        ],
+      },
+    ],
+  },
+  overrides: [
+    {
+      files: ['./src/domain/**/*'],
+      rules: {
+        '@typescript-eslint/no-restricted-imports': [
+          'error',
+          {
+            name: '@neofinancial/neo-framework',
+            importNames: ['!BaseRepositoryPort'],
+            message: 'Hex: domain layer cannot know infrastructure details (MongoDB)',
+          },
+          {
+            name: 'mongodb',
+            message: 'Hex: domain layer cannot know infrastructure details (MongoDB)',
+          },
+          {
+            name: 'mongoose',
+            message: 'Hex: domain layer cannot know infrastructure details (MongoDB)',
+          },
+          {
+            name: '@neofinancial/neo-queue',
+            message: 'Hex: domain layer cannot know infrastructure details (queues)',
+          },
+          {
+            name: 'aws-sdk',
+            message: 'Hex: domain layer cannot know infrastructure details (AWS)',
+          },
+          {
+            name: '@neofinancial/neo-qldb',
+            message: 'Hex: domain layer cannot know infrastructure details (QLDB)',
+          },
+          {
+            name: '@neofinancial/neo-redis-client',
+            message: 'Hex: domain layer cannot know infrastructure details (Redis)',
+          },
+          {
+            name: '@neofinancial/neo-s3',
+            message: 'Hex: domain layer cannot know infrastructure details (S3)',
+          },
+          {
+            name: '@neofinancial/neo-elasticsearch',
+            message: 'Hex: domain layer cannot know infrastructure details (Elasticsearch)',
+          },
```

Comment:
This isn't really an exhaustive list, makes me wonder if there's a way to specify "Don't import from any dependencies" but then just allow people to define the exceptions in their own service's rule overrides. For the odd package that you actually need the types from

Reply:
couldn't find any rules from the import plugin for restricting imports from installed dependencies 🤔